sdk/openai/openai/sources/customizations/api/getSSEs.ts

```diff
@@ -3,18 +3,51 @@
 
 import { StreamableMethod } from "@azure-rest/core-client";
 import { EventMessage, iterateSseStream } from "@azure/core-sse";
-
-async function getStream<TResponse>(
-  response: StreamableMethod<TResponse>
-): Promise<AsyncIterable<Uint8Array>> {
-  const stream = (await response.asNodeStream()).body;
-  if (!stream) throw new Error("No stream found in response. Did you enable the stream option?");
-  return stream as AsyncIterable<Uint8Array>;
-}
+import { RestError } from "@azure/core-rest-pipeline";
 
 export async function getSSEs(
   response: StreamableMethod<unknown>
 ): Promise<AsyncIterable<EventMessage>> {
   const chunkIterator = await getStream(response);
   return iterateSseStream(chunkIterator);
 }
+
+async function getStream<TResponse>(
+  response: StreamableMethod<TResponse>
+): Promise<AsyncIterable<Uint8Array>> {
+  const { body, status } = await response.asNodeStream();
+  if (status !== "200" && body !== undefined) {
+    const text = await streamToText(body);
+    throw JSON.parse(text).error;
```

Comment:
Would we want to wrap this message or whatever in an error object or custom error object?